.circleci/config.yml

```diff
@@ -395,207 +395,319 @@ jobs:
       name: ubuntu
       machine: xlarge
     steps:
-      - checkout
-      - add_ssh_keys:
-          fingerprints:
-            - "3b:cd:47:bf:57:9c:e5:36:b0:4d:5f:12:5e:d3:b3:3e"
-      - restore-gradle-cache
-      - init-aws
-      - run:
-          name: ktlint
-          command: ./gradlew ktlint
-      - run:
-          name: license
-          command: python scripts/license-validate.py
-      - run:
-          name: check changelog
-          command: |
-            if [[ ! -z "${CIRCLE_PULL_REQUEST}" ]]; then
-              echo "$(./mbx-ci github reader token)" > gh_token.txt
-              ./scripts/check-changelog.sh
-            fi
-      - save-gradle-cache
-      - store-results
+      - when:
+          condition:
```

Comment:
will it be enough to add condition for this job? all other jobs depends on it

Reply:
all other jobs have the same condition

Reply:
I mean that `verify-code` is the initial point for all other jobs. And if you filter it, then all other jobs will not start.

Reply:
since our PR merge strategy requires the other jobs, these required jobs has to pass 🙈

Reply:
wonder if there's a way to simplify these conditions, could we add a base job/workflow?

Reply:
All jobs execute because they have alternative step **exit 0**

Reply:
I don't know have to avoid empty job runs because we have a strong branch rule

Reply:
IMHO, it might be good to add a comment explaining why the `exit 0` is needed. Otherwise, I'm afraid we will forget it (especially when adding new jobs)